test/index.test.ts

```diff
@@ -1,215 +1,236 @@
 import { WktUserOptions, geoJSONToWkt, wktToGeoJSON } from "./index";
-import { test, describe, it, expect } from "vitest";
+import { describe, it, expect } from "vitest";
 import { Geometry } from "geojson";
 import proj4 from "proj4";
-import { WKT_GEOMETRY_TYPES } from "../constants";
+import { EMPTY, WKT_GEOMETRY_TYPES, WKTGeometryType, ZZM } from "../constants";
 
-function reversible(
-  wkt1: string,
-  geometry1: Geometry,
-  userOptions: WktUserOptions = {
-    emptyAsNull: true,
-  }
-) {
-  const geometry = wktToGeoJSON(wkt1, userOptions);
-  expect(geometry).not.toBeNull();
-  const wkt = geoJSONToWkt(geometry!);
-  expect(geometry).toEqual(geometry1);
-  expect(wkt).toEqual(wkt1);
+const WKT_GEOMETRY_TYPES_MAP = WKT_GEOMETRY_TYPES.reduce((map, type) => {
+  map[type] = type;
+  return map;
+}, {} as Record<WKTGeometryType, string>);
+
+const dictionary = {
+  Srid: "SRID",
+  Z: ZZM[1],
+  Empty: EMPTY,
 }
+type Dictionary = typeof dictionary;
 
-describe("parsing and stringifying", () => {
-  it("overflow resistance", () => {
-    const bigString = `MULTIPOLYGON (((${Array.from(
-      { length: 1000 },
-      (_, i) => `${i}, ${i}`
-    ).join(",")})))`;
-    expect(wktToGeoJSON(bigString)).toHaveProperty("type", "MultiPolygon");
-  });
-  it("default as null on empty", () => {
-    for (const name of WKT_GEOMETRY_TYPES) {
-      expect(wktToGeoJSON(`${name} EMPTY`)).toBeNull();
-      expect(
-        wktToGeoJSON(`${name} EMPTY`, {
-          emptyAsNull: false,
-        })
-      ).toBeTruthy();
+export function testCases(wktGeometriesMap: Record<WKTGeometryType, string>, dictionary: Dictionary) {
```

Comment:
I think that the code changes are great, but these test changes are not ideal. This is adding a bunch of complexity to the tests and makes them harder to read - I can't really scan tests and know what the inputs & outputs are anymore, without thinking back to what the dictionaries are doing, and this isn't using a vitest abstraction for doing the uppercase & lowercase test cases, so the test specs don't have useful descriptions (they're all "parsing and stringifying").

Given that the additional code change is pretty slim, I think just an additional test or two with uppercase and lowercase versions of inputs would be enough to confirm that the logic works & get full test coverage. And that would mean basically the same test code as before, with a few extra cases.

Reply:
I wanted to harvest the complexity of the current test suite and keep it easily maintainable.
But those are some valid points. And it should be probably sufficient to test some (more complex) cases.